src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1213,7 +1214,19 @@ static bool addTriggeredSubscriptions_withCache
       LM_T(LmtMongo, ("adding subscription: '%s'", sub.toString().c_str()));
 
       long long throttling       = sub.hasField(CSUB_THROTTLING) ? getField(sub, CSUB_THROTTLING).numberLong() : -1;
-      long long lastNotification = sub.hasField(CSUB_LASTNOTIFICATION) ? getIntField(sub, CSUB_LASTNOTIFICATION) : -1;
+      long long lastNotification = -1;
+
+      if (sub.hasField(CSUB_LASTNOTIFICATION))
```

Comment:
What I'm looking for is to use a function that will actually warn about a change in type.
If append() methods of BSONObjBuilder (I imagine there is more than one) accept Int **or** Long, then we have the same problem again - we will not be warned.

So C++ ... elegant perhaps, but **so** dangerous ...